cockatrice/src/gameselector.cpp

```diff
@@ -212,3 +216,17 @@ void GameSelector::processGameInfo(const ServerInfo_Game &info)
 {
     gameListModel->updateGameList(info);
 }
+
+void GameSelector::updateButtonChoices(const QModelIndex &index)
+{
+    if (!index.isValid())
+        return;
+
+    const ServerInfo_Game &game = gameListModel->getGame(index.data(Qt::UserRole).toInt());
+    bool overrideRestrictions = !tabSupervisor->getAdminLocked();
```

Comment:
If you want to check for permissions, isnt it better to do it through the user `ServerInfo_User *getUserInfo() const { return userInfo; }` from `TabSupervisor`, rather than checking if a tab is locked?